node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeList.java

```diff
@@ -133,6 +146,16 @@ public NodeList state(Collection<Node.State> nodeStates) {
         return matching(node -> nodeStates.contains(node.state()));
     }
 
+    /** Returns the subset of nodes in the active state */
+    public NodeList active() {
```

Comment:
Imho the only purpose of this class is to make node filtering statements shorter, so why not go all in.

Reply:
Aside from polluting the interface, it also has disadvantage with code search. I often search for usage of those enum values to see which cases I might need to consider. If we end up having a method for every/most states, that just slows that process down where you would also need to search for usage of those methods.

Reply:
Don't agree with those and the latter is really an argument against all library code, but as a compromise I removed active() since it was used once, but not hosts() since it was used twice :-)